vm_eval.c

```diff
@@ -885,41 +885,47 @@ NORETURN(static void raise_method_missing(rb_execution_context_t *ec, int argc,
 
 /*
  *  call-seq:
- *     obj.method_missing(symbol [, *args] )   -> result
- *
- *  Invoked by Ruby when <i>obj</i> is sent a message it cannot handle.
- *  <i>symbol</i> is the symbol for the method called, and <i>args</i>
- *  are any arguments that were passed to it. By default, the interpreter
- *  raises an error when this method is called. However, it is possible
- *  to override the method to provide more dynamic behavior.
- *  If it is decided that a particular method should not be handled, then
- *  <i>super</i> should be called, so that ancestors can pick up the
- *  missing method.
- *  The example below creates
- *  a class <code>Roman</code>, which responds to methods with names
- *  consisting of roman numerals, returning the corresponding integer
- *  values.
- *
- *     class Roman
- *       def roman_to_int(str)
- *         # ...
- *       end
+ *    method_missing(method, *args) -> object
  *
- *       def method_missing(symbol, *args)
- *         str = symbol.id2name
- *         begin
- *           roman_to_int(str)
- *         rescue
- *           super(symbol, *args)
- *         end
- *       end
- *     end
+ *  Invoked when +self+ called with a method it does not define;
+ *  raises NoMethodError:
+ *
+ *    'foo'.nosuch
+ *    # Raises NoMethodError: undefined method `nosuch' for an instance of String
+ *
+ *  The method may be overridden to provide dynamic handling for the call;
+ *  the overriding method should call +super+ if it can't handle the call.
+ *
+ *  This class overrides +method_missing+:
+ *
+ *    require 'roman-numerals' # Use Ruby gem roman-numerals.
```

Comment:
I'm not a fan of having external dependencies in the example. Maybe sometime simple like this is a good enough example of now `method_missing` works:

```ruby
class Greeter
  def method_missing(method_name, *args, &block)
    if method_name.to_s =~ /^greet_(.*)$/
      puts "Hello #{$1.capitalize}"
    else
      super
    end
  end
end

greeter = Greeter.new
greeter.greet_world # => Hello world
greeter.greet_ruby # => Hello ruby
greeter.foo # Raises NoMethodError: undefined method 'foo' for an instance of Greeter
```